File: Cores/ExtremumScalaCore/src/main/scala/OSOL/Extremum/Core/Scala/Random/Distributions/DiscreteUniform.scala
package OSOL.Extremum.Core.Scala.Random.Distributions

trait DiscreteUniform {

  def getDiscreteUniform(min: Int, max: Int): Int

  final def getDiscreteUniform(area: Map[String, (Int, Int)]): Map[String, Int] =
    area.map { case (key, (min, max)) => (key, getDiscreteUniform(min, max)) }

}

File: Cores/OSOL.Extremum.Core.DotNet/Random/Distributions/DiscreteUniform.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace OSOL.Extremum.Core.DotNet.Random.Distributions
{
    public interface DiscreteUniform
    {
        int getDiscreteUniform(int min, int max);   
    }

    public static class DiscreteUniformFunctions
    {
        public static Dictionary<string, int> Diameter(this DiscreteUniform GoRN, Dictionary<string, Tuple<int, int>> area) =>
            area.ToDictionary(kvp => kvp.Key,
                kvp => GoRN.getDiscreteUniform(kvp.Value.Item1, kvp.Value.Item2));
    }

}